docs/user_guides/fs/tags/tags.md

```diff
@@ -0,0 +1,138 @@
+      <img src="../../../../assets/images/guides/tags/tags_schema_simple.png" alt="UI tag schema definition">
+    </a>
+    <figcaption>UI tag schema definition</figcaption>
+  </figure>
+</p>
+
+The UI tool allows to define simple not-nested schemas. For more advanced use cases, more complex schemas (e.g. nested schemas) might be required to fully express the content of a given artifact.
+In such cases it is possible to provide the schema directly as JSON string. The JSON should follow the standard [https://json-schema.org](https://json-schema.org). An example of complex schema is the following:
+
+```
+{
+  "type" : "object",
+  "properties" :
+  {
+    "first_name" : { "type" : "string" },
+    "last_name" : { "type" : "string" },
+    "age" : { "type" : "integer" },
+    "hobbies" : {
+        "type" : "array",
+        "items" : { "type" : "string" }
+    }
+  },
+  "required" : ["first_name", "last_name", "age"],
+  "additionalProperties": false
+}
+```
+
+Additionally it is also possible to define single property as tag. This is achieved by defining a JSON schema like the following:
+
+```
+{ "type" : "string" }
+```
+
+Where the type is a valid primitive: `string`, `boolean`, `integer`, `number`.
+
+## Step 2: Attach a tag to an artifact
+
+Once the schema has been created, you can attach a tag with that schema to a feature group, feature view or training dataset either using the feature store APIs, or by using the UI.
+
+### Using the API
+
+You can attach tags to feature groups and feature views by using the `add_tag()` method of the feature store APIs:
+
+=== "Python"
+
+    ```python
+    # Retrieve the feature group
+    fg = fs.get_feature_group("transactions_4h_aggs_fraud_batch_fg", version=1)
+
+    # Define the tag
+    tag = {
+        'business_unit': 'Fraud',
+        'data_owner': 'email@hopsworks.ai',
+        'pii': True
+    }
+
+    # Attach the tag
+    fg.add_tag("data_privacy", tag) 
+    ```
+
+You can see the list of tags attached to a given artifact by using the `get_tags()` method:
+
+=== "Python"
+
+    ```python
+    # Retrieve the feature group
+    fg = fs.get_feature_group("transactions_4h_aggs_fraud_batch_fg", version=1)
+
+    # Retrieve the tags for this feature group 
+    fg.get_tags()
+    ```
+
+Finally you can remove a tag from a given artifact by calling the `delete_tag()` method:
+
+=== "Python"
+
+    ```python
+    # Retrieve the feature group
+    fg = fs.get_feature_group("transactions_4h_aggs_fraud_batch_fg", version=1)
+
+    # Retrieve the tags for this feature group 
+    fg.delete_tag("data_privacy")
+    ```
+
+The same APIs work for feature views and training dataset alike.
+
+### Using the UI
+
+You can attach tags to feature groups and feature views directly from the UI. You can navigate on the artifact page and click on the `Add tags` button. From there you can select the schema of the tag you want to attach and populate the values as shown in the gif below.
```

Comment:
You can navigate on the artifact page and click on the `Add tags` button.
- I am looking at the UI, can't see how "Add tags" for FGs (it's inside edit), FVs and TDs (nada)
I assume there is PR making all of this consistent?

"select the tag schema"

Reply:
From there,

Reply:
Latest master has the `Add Tags` button for both FGs and FVs. TDs have a edit tags button.